nomos-libp2p/src/lib.rs

```diff
@@ -51,14 +66,58 @@ impl Behaviour {
                 .max_transmit_size(DATA_LIMIT)
                 .build()?,
         )?;
-        Ok(Self { gossipsub })
+
+        // make config parameters configurable
+        let store = kad::store::MemoryStore::new(peer_id);
```

Comment:
Would be great to implement our own storage to add recovery later on.

Reply:
This is a great idea, can we abstract it and still use the `MemoryStore` ourselves so its easier to move later on?